### ISSUE TITLE
Switching between Crowdmap and Crowdmap Classic

### ISSUE THREAD

**Issue by heatherleson** [ushahidi/crowdmap-issues#106]
There is irregular cache and workflow behaviour switching between Crowdmap and Crowdmap Classic

User experience:
[13-05-01 9:34:58 AM] Heather Leson: invite code.
[13-05-01 9:35:28 AM] Heather Leson: crowdmap.com. scroll to bottom of screen, click on "new version", set up account with xxxxxx (invite code removed)
[13-05-01 9:37:14 AM] Heather Leson: wiki article https://wiki.ushahidi.com/display/WIKI/Switching+between+Crowdmap+and+Crowdmap+Classic
[13-05-01 9:39:56 AM] xxx: Great, thanks Heather!
[13-05-01 9:40:05 AM] Heather Leson: let me know how it works
[13-05-01 9:41:00 AM] xxxx: will do - mightn't get round to much testing until later this coming weekend/early next week, mind
[13-05-01 9:41:22 AM] Heather Leson: no worries. I am more worried about the sign up process and switching between accounts
[13-05-01 10:22:27 AM] xxx: "That invitation does not appear to be valid." :(
[13-05-01 10:22:37 AM] Heather Leson: hmm
[13-05-01 10:23:02 AM] xxxx: maybe it's copying from Skype issue...
[13-05-01 10:23:15 AM] xxxx: may have introduced some spurious characters...
[13-05-01 10:23:24 AM] xxxx: let me try again...
[13-05-01 10:23:27 AM] Heather Leson: (invite code removed)
[13-05-01 10:23:30 AM] Heather Leson: let me know.
[13-05-01 10:26:51 AM] xxxx: hmmm...
[13-05-01 10:27:19 AM] Heather Leson: so no dice?
[13-05-01 10:27:21 AM] xxxx: this is getting frustrating - I seem to be in some kind of loop where it's getting confused by my existing crowdmap a/c...
[13-05-01 10:27:32 AM] Heather Leson: ok
[13-05-01 10:27:39 AM] Heather Leson: next. which browser are you using?
[13-05-01 10:27:54 AM] xxxx: and either tells me that a new username or password is wrong...or says my email already taken...or username taken etc...
[13-05-01 10:27:56 AM] Heather Leson: there was a cookie cache thing. they said they fixed. i have a workaround
[13-05-01 10:27:59 AM] xxx: Firefox
[13-05-01 10:28:04 AM] Heather Leson: ok. do this
[13-05-01 10:28:18 AM] Heather Leson: http://support.mozilla.org/en-US/kb/delete-cookies-remove-info-websites-stored
[13-05-01 10:28:24 AM] Heather Leson: removed only crowdmap.com ones.
[13-05-01 10:28:37 AM] Heather Leson: i will tell them that it still exists for existing users
[13-05-01 10:28:52 AM] Heather Leson: are you trying to login with your existing crowdmap account?
[13-05-01 10:30:44 AM] Heather Leson: it shows you have accepted 

(username deleted) accepted your beta invitation.
 : P
[13-05-01 10:30:50 AM] xxx: I was trying to set up new one...but it seemed to want me to set up new one...but I suppose then it would prefer if I used a new email address...which I'd prefer not to (try and keep my activities reasonably segregated for various addresses - otherwise it gets very confusing)
[13-05-01 10:30:56 AM] Heather Leson: hmm
[13-05-01 10:30:59 AM] Heather Leson: adding to a ticket
[13-05-01 10:31:16 AM] xxx: OK - so (Other username deleted)'s accepted.
[13-05-01 10:31:29 AM] Heather Leson: but not the other username - which one?
[13-05-01 10:31:35 AM] xxxx: see if getting rid of cookies will allow me to activate that identity..
[13-05-01 10:31:40 AM] xxx: (username removed)!
[13-05-01 10:37:17 AM] Heather Leson: https://github.com/ushahidi/crowdmap-issues/issues/105
[13-05-01 12:34:03 PM] xxxx: hmmm, right - I clicked on the activation email for (username deleted)...gave me messages about not recognising the invitation code....had to sign in (using old crowdmap a/c password :^))
[13-05-01 12:34:08 PM] xxxx: and I'm now at https://crowdmap.com/mhi/manage
[13-05-01 12:34:14 PM] Heather Leson: hi
[13-05-01 12:34:18 PM] xxxx: does that mean I'm 'in'?
[13-05-01 12:34:29 PM] Heather Leson: did the screens change?
[13-05-01 12:35:12 PM] xxx: I can see 'Your deployments', 'account settings' buttons at the front
[13-05-01 12:35:20 PM] xxxx: should I see more?
[13-05-01 12:35:26 PM] Heather Leson: you should not see that
[13-05-01 12:35:40 PM] xxxx: It does say Crowdmap beta...and has different 'livery'
[13-05-01 12:37:12 PM] Heather Leson: you should see - pages like these - https://wiki.ushahidi.com/display/WIKI/Crowdmap+-+Basic+Features
[13-05-01 12:38:59 PM] Heather Leson: so to recap 1. removed cookies 2. signed up for beta  3. logged back into crowdmap with your usual crowdmap user id
[13-05-01 12:39:09 PM] Heather Leson: do you see "new version" at the bottom of the page
[13-05-01 12:39:22 PM] Heather Leson: try clicking that
[13-05-01 12:39:35 PM] Heather Leson: also, I can't thank you enough for this
[13-05-01 12:49:15 PM] Heather Leson: i was able to switch back from the login menu
[13-05-01 12:53:27 PM] xxxx: I still see New version at the bottom...but otherwise it looks sortof good
[13-05-01 12:53:49 PM] Heather Leson: ok. so when you click on new version?
[13-05-01 12:54:13 PM] xxxx: yeah :) I think I'm in now
[13-05-01 12:54:22 PM] Heather Leson: ok, did you have to relogin ?
[13-05-01 12:54:25 PM] xxxx: (after clicking new version)
[13-05-01 12:54:30 PM] Heather Leson: ok. thank you
[13-05-01 12:54:29 PM] xxxx: no
[13-05-01 12:54:40 PM] Heather Leson: i am going to share this with our developers.
[13-05-01 12:54:47 PM] Heather Leson: user experience so key
[13-05-01 12:54:48 PM] Heather Leson: :)
[13-05-01 12:54:52 PM] xxx: grand - happy to help!
[13-05-01 12:54:55 PM] Heather Leson: thanks for your patience. happy tinkering
[13-05-01 12:55:03 PM] xxxx: nae bother!
[13-05-01 12:55:34 PM] Heather Leson: for fun - crowdmap.com/photos : )
[13-05-01 12:55:47 PM] Heather Leson: you can see some awesome pics from instagram etc

**Comment by evansims** [comment]
Whew, OK. I'll try to break this down as best I can.

```
[13-05-01 10:27:21 AM] xxxx: this is getting frustrating - I seem to be in some kind of loop where it's getting confused by my existing crowdmap a/c...
```

I really wish I knew more about what they meant by this, as it seems like the critical point of confusion.

Crowdmap Classic accounts are transitioned transparently when they login using their email address and password. You can either register for a New account using the same email and password you use on Classic (if you don't you'll be informed that email is already in use) or login using the same credentials (at which point you'll be asked to select a username, etc.)

```
[13-05-01 10:27:56 AM] Heather Leson: there was a cookie cache thing. they said they fixed. i have a workaround
[13-05-01 10:28:18 AM] Heather Leson: http://support.mozilla.org/en-US/kb/delete-cookies-remove-info-websites-stored
```

Sorry for the confusion — We don't want to suggest users delete cookies with New Crowdmap issues. The corrupt cookie bug only applies to Classic/Platform and doesn't have a role in New issues. The users deleting of the cookies is why they started hitting Classic's manage page after that point; it wiped out the cookie that tells them to hit our 'New' servers instead of 'Classic' (which is where everyone lands by default.)

```
[13-05-01 10:30:44 AM] Heather Leson: it shows you have accepted
(username deleted) accepted your beta invitation.
```

Their account creation was successful. Further attempts to create accounts would fail past this point, naturally.

```
[13-05-01 10:30:50 AM] xxx: I was trying to set up new one...but it seemed to want me to set up new one...but I suppose then it would prefer if I used a new email address...which I'd prefer not to (try and keep my activities reasonably segregated for various addresses - otherwise it gets very confusing)
```

Email accounts are tied to usernames; you cannot have multiple accounts with the same email address. (Tip: A nice workaround for this if you're using Gmail or Google Apps for Domains is to apply a +label to your address, immediately after the username. i.e. evan+spammers@ushahidi.com; it will still go to your Inbox, but most authentication systems, including ours, will identify it as a seperate email address.)

```
[13-05-01 12:34:03 PM] xxxx: hmmm, right - I clicked on the activation email for (username deleted)...gave me messages about not recognising the invitation code....had to sign in (using old crowdmap a/c password :^))
[13-05-01 12:34:08 PM] xxxx: and I'm now at https://crowdmap.com/mhi/manage
```

They wiped out their cookies, which is why they had to sign back in and landed back at /mhi/manage on Classic.

---

I don't think we're really dealing with any bugs here, but there are some points of confusion that we need to address. How can we better communicate that you should login using your existing Crowdmap email address and password?

**Comment by heatherleson** [comment]
user did attempt with existing id first, but it was listed as invalid. So he setup a new one.

**Comment by aoduor** [comment]
Have a feeling this is linked to special character issues in the password. Heather, any chance you could check with the deployer if he has any special characters in his password? I had this same experience when I was setting up my crowdmap account initially.

Have a look at https://github.com/ushahidi/crowdmap-issues/issues/122

**Comment by brianherbert** [comment]
@heatherleson Make any progress here figuring out if the user was using special characters?

**Comment by evansims** [comment]
Closing as this seems to have been a case of confusion about the login process rather than a bug; we've made updates the login UI to make things a bit clearer. Please open a new ticket if login problems are encountered again. Thanks.
